src/components/AppendStepButton.tsx

```diff
@@ -0,0 +1,90 @@
+import { StepsService, ValidationService } from '@kaoto/services';
+import { useIntegrationJsonStore, useSettingsStore } from '@kaoto/store';
+import { IStepProps } from '@kaoto/types';
+import { Popover, Tooltip } from '@patternfly/react-core';
+import { PlusIcon } from '@patternfly/react-icons';
+import { FunctionComponent, useEffect, useState } from 'react';
+import { BranchBuilder } from './BranchBuilder';
+import { MiniCatalog } from './MiniCatalog';
+
+interface IAddStepButton {
+  handleAddBranch: () => void;
+  handleSelectStep: (selectedStep: IStepProps) => void;
+  step: IStepProps;
+  showBranchesTab: boolean;
+  showStepsTab: boolean;
+  supportsBranching: boolean;
+}
+
+export const AppendStepButton: FunctionComponent<IAddStepButton> = ({
+  handleAddBranch,
+  handleSelectStep,
+  step,
+  showBranchesTab,
+  showStepsTab,
+  supportsBranching,
+}) => {
+  const currentDSL = useSettingsStore((state) => state.settings.dsl.name);
+  const views = useIntegrationJsonStore((state) => state.views);
+  const [hasCustomStepExtension, setHasCustomStepExtension] = useState(StepsService.hasCustomStepExtension(step, views));
+  const [disableBranchesTabMsg, setDisableBranchesTabMsg] = useState('');
+
+  useEffect(() => {
+    setHasCustomStepExtension(StepsService.hasCustomStepExtension(step, views));
+  }, [step, views])
+
+  useEffect(() => {
+    if (hasCustomStepExtension) {
+      setDisableBranchesTabMsg(`The "${step.name}" step has a Custom Step Extension, please click on the step to configure it`);
```

Comment:
I do wonder if the language should be simpler for the end user who doesn't know what a step extension is (my guess is that the person who sets it up would be the developer, and the one to use it would be the non-dev).

For example, maybe just saying `"Please click on the step to configure branches for it."` is enough? However, the user might wonder why this only applies to certain steps.. so I'm not 100% sure. 🤔  @igarashitm @mmelko

Reply:
Ah good point, the word `Step Extension` would be a tech term for Kaoto developers. Can we say `click on the step to configure branches`? Not sure if it's always branches

Reply:
@igarashitm we shouldn't forget the magic word :stuck_out_tongue_closed_eyes:
 "please"

Reply:
I think that you're right, I'll update the pull request to use your example :bow: